Mathlib/Topology/PartialHomeomorph.lean

```diff
@@ -1402,11 +1404,18 @@ lemma lift_openEmbedding_trans_apply
   simp [hf.injective.extend_apply e']
 
 @[simp]
-lemma lift_openEmbedding_trans (e e' : PartialHomeomorph X Z) (hf : IsOpenEmbedding f) :
+lemma lift_openEmbedding_trans
+    (e e' : PartialHomeomorph X Z) (hf : IsOpenEmbedding f) :
     (e.lift_openEmbedding hf).symm.trans (e'.lift_openEmbedding hf) = e.symm.trans e' := by
-  ext x
-  · exact e.lift_openEmbedding_trans_apply e' hf x
-  · simp [hf.injective.extend_apply e]
+  ext z
+  · by_cases h': Nonempty X; swap
+    · have := not_nonempty_iff.mp h'
+      exact (IsEmpty.false (e.symm z)).elim
+    exact e.lift_openEmbedding_trans_apply e' hf z
+  · by_cases h': Nonempty X; swap
+    · have := not_nonempty_iff.mp h'
+      exact (IsEmpty.false (e.symm z)).elim
+    simp [hf.injective.extend_apply e]
```

Comment:
```suggestion
  · exact e.lift_openEmbedding_trans_apply e' hf z
  · simp [hf.injective.extend_apply e]
```
are you sure that the previous code does not work?  It appers to be fine in the online server.

Reply:
In a previous version, it did not work - but I had some superfluous non-emptiness hypothesis which the unusedHave linter ofund, perhaps it was that one. In any case, the code works now: thanks for the suggestion!